tests/test_modules/test_ADCore/test_detectordriverpart.py

```diff
@@ -154,6 +154,32 @@ def test_run(self):
         assert self.o.registrar.report.call_count == 2
         assert self.o.registrar.report.call_args[0][0].steps == 0
 
+    def test_hardware_triggered_run(self):
+        self.o.registrar = MagicMock()
```

Comment:
What is this test testing for?

Reply:
It tests that we do not call arm_detector in run() if we are hardware triggered